deployment/update_production.sh

```diff
@@ -1,37 +1,56 @@
 #!/bin/bash
 set -e # abort on error
-cd `dirname $0`/.. # change to root directory
+cd `dirname $0`/.. # change to project root directory
 
-sudo -H -u evap git fetch
+echo $PWD
+
+# used for constructing the backup file name
+COMMIT_HASH="$(git rev-parse --short HEAD)"
+BACKUP_TITLE="backup"
+TIMESTAMP="$(date +%Y-%m-%d_%H:%M:%S)"
+
+USERNAME="evap"
+ENVDIR="/home/evap/env"
+[[ ! -z "$EVAP_RUNNING_INSIDE_TRAVIS" ]] && echo "Detected travis" && USERNAME="travis" && ENVDIR=~/virtualenv/python3.7
+
+# argument 1 is the title for the backupfile.
+if [ $# -eq 1 ]
+    then
+        BACKUP_TITLE=$1
+fi
+
+FILENAME="${TIMESTAMP}_${COMMIT_HASH}_${BACKUP_TITLE}.json"
+
+[[ ! -z "$EVAP_OVERRIDE_BACKUP_FILENAME" ]] && echo "Overriding Automatic Filename" && FILENAME="${BACKUP_TITLE}"
+
+echo "Backup will be stored in $FILENAME"
+echo "Starting update..."
+
+set -x # print executed commands. enable this here to not print the if above.
+
+sudo -H -u $USERNAME git fetch
 
 # Note that apache should not be running during most of the upgrade,
 # since then e.g. the backup might be incomplete or the code does not
 # match the database layout, or https://github.com/fsr-de/EvaP/issues/1237.
-sudo service apache2 stop
+[[ -z "$EVAP_RUNNING_INSIDE_TRAVIS" ]] && sudo service apache2 stop
 
-# argument 1 is the filename for the backupfile.
-# if no argument is present, no backup will be created.
-if [ $# -eq 1 ] # if there is exactly one argument
-    then
-        echo Creating backup.
-        set -x # print executed commands. enable this here to not print the if above.
-        sudo -H -u evap env/bin/python manage.py dumpdata --indent 2 --output $1
-    else
-        echo No backup file specified, skipping backup creation.
-        set -x # print executed commands
-fi
+sudo -H -u $USERNAME $ENVDIR/bin/python manage.py dumpdata --natural-foreign --natural-primary --all -e contenttypes -e auth.Permission --indent 2 --output $FILENAME
```

Comment:
could you try putting that into dump_testdata?

Reply:
It changes ID handling for the users: The users will not be referenced by id anymore but by username. The dump will not contain the old user id.

Reply:
hm, is that a problem?

Reply:
Nope. Will add these parameters to the dump_testdata command.